### PR TITLE
Support for Instrumentation Testing

### PR DESCRIPTION
**Pull request by ehanoc**
Maybe useful moving forward? I'll be adding some from time to time as i use these for this project and my own projects integrations. 

I would use straight unit tests but we need instrumentation to support native libs (Unless we would mock them).

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -43,6 +43,16 @@ dependencies {
     compile 'com.android.support:cardview-v7:25.4.0'
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
     compile 'me.dm7.barcodescanner:zxing:1.9.8'
+
+    androidTestCompile 'com.android.support:support-annotations:25.4.0'
+    androidTestCompile 'com.android.support.test:runner:0.5'
+    androidTestCompile 'com.android.support.test:rules:0.5'
```
Comment (EarlOfEgo) [inline app/build.gradle:49]: Would makes sense to extract the versions into a constant.

File: app/build.gradle [hunk 2]
```diff
@@ -43,6 +43,16 @@ dependencies {
...
+    // Optional -- Hamcrest library
+    //androidTestCompile 'org.hamcrest:hamcrest-library:1.3'
+    // Optional -- UI testing with Espresso
+    //androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.2'
+    // Optional -- UI testing with UI Automator
+    //androidTestCompile 'com.android.support.test.uiautomator:uiautomator-v18:2.1.2'
```
Comment (EarlOfEgo) [inline app/build.gradle:55]: why committing uncommented code?

File: app/src/androidTest/java/com/m2049r/xmrwallet/util/HelperTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+public class HelperTest {
+
+    /**
+     *
+     */
+    @Test
+    public void checkValidAddressExample_1() {
+        // random generated address by https://xmr.llcoins.net/addresstests.html
+        String publicAddr = "4GKacjHuTGeReA7PvxPhehDswu3RwpnAqeJV2kdU3MLMNUr4Lt9pxcRiWU2E6BoDReADhCPtHfvNFh1YVNLFoZE5YtaSMc8MSbGGudRftK";
+        Assert.assertTrue(Wallet.isAddressValid(publicAddr, false));
```
Comment (EarlOfEgo) [inline app/src/androidTest/java/com/m2049r/xmrwallet/util/HelperTest.java:28]: Why is this test part of `HelperTest`? Shouldn't it be in something like `WalletTest`? In this file I would expect tests that cover the methods of the `Helper` class.

File: app/build.gradle (hunk 1 above)
Comment (ehanoc) [reply app/build.gradle:49]: 👍  makes sense

File: app/build.gradle (hunk 2 above)
Comment (ehanoc) [reply app/build.gradle:55]: We could write-up some guidelines to make it clear. I'm happy to take them off though

File: app/src/androidTest/java/com/m2049r/xmrwallet/util/HelperTest.java (hunk 3 above)
Comment (ehanoc) [reply app/src/androidTest/java/com/m2049r/xmrwallet/util/HelperTest.java:28]: It's a simple dummy test, just an example part of the integration for the tests. To be fair Wallet is mostly JNI mapping to the actual native wallet lib. I think it already has it's own tests in the monero tree. We could / should mock these in the future and focus the tests on monerujo's scope. 

I'm already writing my tests for my project & fork. I can delay submitting all this later if it's something we want on this project

File: app/build.gradle (hunk 2 above)
Comment (EarlOfEgo) [reply app/build.gradle:55]: I agree, Guidelines would be very helpful. In general I'm against uncommented code, imo if it is uncommented it is not needed.

File: app/src/androidTest/java/com/m2049r/xmrwallet/util/HelperTest.java (hunk 3 above)
Comment (EarlOfEgo) [reply app/src/androidTest/java/com/m2049r/xmrwallet/util/HelperTest.java:28]: If you write tests and submit them when you are done that would be amazing.

File: app/build.gradle (hunk 2 above)
Comment (m2049r) [reply app/build.gradle:55]: yes, please no commented-out code

**Comment by EarlOfEgo** [comment]
I suggest we close the PR and when @ehanoc is ready with his tests he opens a new one. I think there also will be some manual merging need to be done then.

**Comment by m2049r** [comment]
thank you @ehanoc, i agree with @EarlOfEgo

**Pull request closed without merging**